workflow/controller/workflowpod.go

```diff
@@ -834,6 +834,9 @@ func (woc *wfOperationCtx) addArchiveLocation(pod *apiv1.Pod, tmpl *wfv1.Templat
 			break
 		}
 	}
+	if woc.artifactRepository.AutoArchiveLogs != nil && *woc.artifactRepository.AutoArchiveLogs {
+		needLocation = true
+	}
```

Comment:
Hmmm, currently it's not clear that even when enabling `AutoArchiveLogs` the user will also need to enable `ArchiveLogs` for the logs to be archived. Maybe a couple of lines below should actually be:

```go
	tmpl.ArchiveLocation = &wfv1.ArtifactLocation{
		ArchiveLogs: woc.artifactRepository.ArchiveLogs || woc.artifactRepository.AutoArchiveLogs,
	}
```

Reply:
Agreed on clarity. I think your proposal sounds reasonable.